ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -689,6 +695,67 @@ public void shouldNotIncludeSslPropertiesInListPropertiesOutput() {
         not(hasItems(KsqlConfig.SSL_CONFIG_NAMES.toArray(new String[0]))));
   }
 
+  @Test
+  public void shouldNotWaitIfNoCommandTopicOffsetSpecified() {
+    // Given:
+    final String sql = "LIST REGISTERED TOPICS;";
+    final KsqlRequest request = new KsqlRequest(sql, Collections.emptyMap(), null);
+
+    EasyMock.expect(commandStore.getConsumerPositionFuture(EasyMock.anyLong()))
+        .andThrow(new AssertionFailedError()).anyTimes();
+    EasyMock.replay(commandStore);
+
+    // When:
+    makeSingleRequest(request, KsqlTopicsList.class);
+
+    // Then:
+    EasyMock.verify(commandStore);
+  }
+
+  @Test
+  public void shouldNotWaitIfCommandTopicOffsetReached()
```

Comment:
Strangely named test. This shows the code does wait, because its calling `.get()` on the future.